back/app/services/areas_filtering_service.rb

```diff
@@ -0,0 +1,21 @@
+class AreasFilteringService
+  include Filterer
+
+  add_filter('only_selected') do |scope, options|
+    params = options[:params]
+    current_user = options[:current_user]
+
+    next scope unless ['true', true, '1'].include? params[:only_selected]
+
+    publications = AdminPublicationPolicy::Scope.new(current_user, AdminPublication).resolve.includes(:parent)
```

Comment:
I feel like the logic written here , doesn't have much to do with areas, but is more about visibility of admin publications. For example, if you would add a similar filter for topics, wouldn't you have to copy this code over to also leave out topics of folders with no public projects? 

I suspect the API request to the areas endpoint will happen simultaneously with an API request to get all the admin publications (on which the area filter can be applied). Wouldn't both endpoints do the same kind of filtering in that case? If so, I think it would make more sense to move this code to something like an AdminPublicationService, which could be reused for both endpoints (and future ones). 

I think what would be really great is to have a separate service for everything around admin publication visibility (this could be a new method in the UserRoleService for example), that could be called anywhere in the code where we need to return admin publications, projects or folders that only the current user can see.

Reply:
@sebastienhoorens   
  
> I feel like the logic written here , doesn't have much to do with areas, but is more about visibility of admin publications. For example, if you would add a similar filter for topics, wouldn't you have to copy this code over to also leave out topics of folders with no public projects?
  
Thanks. I felt the exact tension you describe when writing this code. For example, I originally wrote an endpoint in the AdminPublicationsController, and then abandoned that as I was in effect returning a list of areas, so it felt wrong to have it in that controller.  And, yes, this pattern would need repeating when filtering by topic(s).

I like both ideas you suggest; an AdminPublicationService or a new method in the UserRoleService, though I can't quite imagine how this could/would actually be used by both endpoints (`admin_publications` and `areas`), and future ones. Could you give me an example request url with imaginary query params that would make use of such a service? Or maybe  we could have a short call next week to discuss/explain?

Reply:
So the code here could become something like:
```
visible_publications = AdminPublicationService.visible_admin_publications_for(AdminPublication, current_user, include_empty_publications: false)
visible_projects = Project.where(id: visible_publications.where(publication_type: Project.name).pluck(:publication_id))
scope.where(id: AreasProject.where(project: visible_projects).pluck(:area_id))
```

This would also filter out draft projects and folders for non-admins. If admins shouldn't see drafts either, you could pass `AdminPublication.published` instead of `AdminPublication` to `visible_admin_publications_for`.

The method `visible_admin_publications_for` could have additional options, such as leaving out folders which would have no children left, or perhaps even a `publication_status` parameter, or something else that might make sense to differentiate between different situations where publications are listed (e.g. listing publications to be edited vs. listing published publications).

I think this method could then also be reused to rewrite the `AdminPublicationsFilteringService` and perhaps other parts of code that exists today.

Reply:
How would it be used by the `areas` endpoint?

Reply:
Also, not seeing 'draft' == seeing only 'published' && 'archived', so `AdminPublication.published` would not be sufficient (but maybe we could have a `not_draft` method).

Reply:
We would need to pass womthing like`AdminPublication.not_draft` to get the base collection required for the homepage, as even Admins are not supposed to see draft items, but both _archived and published_ could be visible.